### PR TITLE
Implement limited access to gateway modes for Adam

### PR DESCRIPTION
**Pull request by bouwew**
Implements feature request #523

### CODE REVIEW

File: plugwise/__init__.py [hunk 1]
```diff
@@ -862,6 +866,30 @@ async def set_switch_state(
...
+        """Set the gateway mode."""
+        if mode not in self._gw_allowed_modes:
+            raise PlugwiseError("Plugwise: invalid gateway mode.")
+
+        time_1 = dt.datetime.now(dt.UTC)
+        away_time = time_1.isoformat(timespec="milliseconds") + "Z"
+        time_2 = str(dt.date.today() - dt.timedelta(1))
+        vacation_time = time_2 + "T23:00:00.000Z"
+        end_time = "2037-04-21T08:00:53.000Z"
+        if mode == "away":
```
Comment (CoMPaTech) [inline plugwise/__init__.py:879]: This looks a bit weird, should we set `valid == ''` by default and then one if/else construction

**Comment by bouwew** [comment]
Not fixing the remaining Sonarcloud issue. Will be fixed when we split the code for legacy and non-legacy.

**Review by CoMPaTech — APPROVED** [review]
We should review the if statements, but other than that, seems solid to me.

**Comment by CoMPaTech** [comment]
> Not fixing the remaining Sonarcloud issue. Will be fixed when we split the code for legacy and non-legacy.

I just login to sonar and set it to 'won't fix' with remark `Known issue` - that relieves the complexity (for now)

File: plugwise/__init__.py (hunk 1 above)
Comment (bouwew) [reply plugwise/__init__.py:880]: Implemented :)

**Review by CoMPaTech — APPROVED** [review]
lgtm

**Pull request merged**